packages/convai-widget-core/src/markdown/components/InfoCard.tsx

```diff
@@ -0,0 +1,92 @@
+import type { ComponentChildren } from "preact";
+import { cn } from "../../utils/cn";
+import { type HTMLAttributes, type ButtonHTMLAttributes } from "preact/compat";
+import { useSignal } from "@preact/signals";
+import { IconName } from "../../components/Icon";
+import { Button } from "../../components/Button";
+import { Signalish } from "../../utils/signalish";
+import { InOutTransition } from "../../components/InOutTransition";
+
+export interface ActionConfig {
+  icon: IconName;
+  label: string;
+  onClick: ButtonHTMLAttributes<HTMLButtonElement>["onClick"];
+  disabled?: Signalish<boolean>;
+  "aria-label"?: string;
+}
+
+type InfoCardActionsProps = {
+  actions: ActionConfig[];
```

Comment:
This should just accept `ComponentChildren` and have the components pass buttons here in a fragment
`ActionConfig` is just reimplementing JSX tbh
```suggestion
  actions: ComponentChildren;
```